indico/modules/rb/client/js/common/bookings/BookingDetails.jsx

```diff
@@ -529,19 +543,25 @@ class BookingDetails extends React.Component {
       actions: {changeBookingState},
     } = this.props;
     const {acceptanceFormVisible} = this.state;
-    this.setState({actionInProgress: action});
+    this.setState({actionInProgress: action, error: null});
     return changeBookingState(id, action, data).then(({error}) => {
-      if (error && error.message === 'prebooking_collision' && action === 'approve') {
-        if (acceptanceFormVisible) {
-          this.hideAcceptanceForm();
+      if (error) {
+        if (error.message === 'prebooking_collision' && action === 'approve') {
+          if (acceptanceFormVisible) {
+            this.hideAcceptanceForm();
+          }
+          this.setState({
+            preBookingConflicts: {
+              concurrentBookings: error.data,
+              acceptanceData: data,
+              warningVisible: true,
+            },
+          });
+        } else {
+          this.setState({
+            error,
+          });
```

Comment:
Do we need to clear `preBookingConflicts` here?

Reply:
As far as I can see, `preBookingConflicts` is reset once the corresponding warning modal is closed. We could reset it just for the sake of being explicit, but since the two state slices never interact, I wouldn't bother...